test/powershell/engine/ExperimentalFeature/assets/ExpTest/ExpTest.psm1

```diff
@@ -0,0 +1,206 @@
+        [string] $Destination,
+
+        [Experimental("ExpTest.FeatureOne", [ExperimentAction]::Hide)]
+        [Parameter(ParameterSetName = "UrlSet")]
+        [Parameter(ParameterSetName = "RadioSet")]
+        [string] $Configuration
+    )
+}
+
+#endregion
+
+#region "Dynamic parameters"
+
+function Test-MyDynamicParamOne
+{
+    [CmdletBinding()]
+    param(
+        [string] $Name
+    )
+
+    ## Use the parameter attribute to hide or show a dynamic parameter.
+    DynamicParam {
+        if ($Name -eq "Joe") {
+            $runtimeParams = [RuntimeDefinedParameterDictionary]::new()
+
+            $configFileAttributes = [System.Collections.ObjectModel.Collection[Attribute]]::new()
+            $configFileAttributes.Add([Parameter]::new("ExpTest.FeatureOne", [ExperimentAction]::Show))
+            $configFileAttributes.Add([ValidateNotNullOrEmpty]::new())
+            $configFileParam = [RuntimeDefinedParameter]::new("ConfigFile", [string], $configFileAttributes)
+
+            $configNameAttributes = [System.Collections.ObjectModel.Collection[Attribute]]::new()
+            $configNameAttributes.Add([Parameter]::new("ExpTest.FeatureOne", [ExperimentAction]::Hide))
+            $configNameAttributes.Add([ValidateNotNullOrEmpty]::new())
+            $ConfigNameParam = [RuntimeDefinedParameter]::new("ConfigName", [string], $configNameAttributes)
+
+            $runtimeParams.Add("ConfigFile", $configFileParam)
+            $runtimeParams.Add("ConfigName", $ConfigNameParam)
+            return $runtimeParams
+        }
+    }
+}
+
+function Test-MyDynamicParamTwo
+{
+    [CmdletBinding()]
+    param(
+        [string] $Name
+    )
+
+    ## Use the experimental attribute to hide or show a dynamic parameter.
+    DynamicParam {
+        if ($Name -eq "Joe") {
+            $runtimeParams = [RuntimeDefinedParameterDictionary]::new()
+
+            $configFileAttributes = [System.Collections.ObjectModel.Collection[Attribute]]::new()
+            $configFileAttributes.Add([Experimental]::new("ExpTest.FeatureOne", [ExperimentAction]::Show))
+            $configFileAttributes.Add([Parameter]::new())
+            $configFileAttributes.Add([ValidateNotNullOrEmpty]::new())
+            $configFileParam = [RuntimeDefinedParameter]::new("ConfigFile", [string], $configFileAttributes)
+
+            $configNameAttributes = [System.Collections.ObjectModel.Collection[Attribute]]::new()
+            $configNameAttributes.Add([Experimental]::new("ExpTest.FeatureOne", [ExperimentAction]::Hide))
+            $configNameAttributes.Add([Parameter]::new())
+            $configNameAttributes.Add([ValidateNotNullOrEmpty]::new())
+            $ConfigNameParam = [RuntimeDefinedParameter]::new("ConfigName", [string], $configNameAttributes)
+
+            $runtimeParams.Add("ConfigFile", $configFileParam)
+            $runtimeParams.Add("ConfigName", $ConfigNameParam)
+            return $runtimeParams
+        }
+    }
+}
+
+#endregion
```

Comment:
You're missing a test cases for:

1. `[Experimental()]`
2. ExperimentAction.None
3. ExperimentName starting and ending with period

Reply:
Tests added.